client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-prefetch-images.ts

```diff
@@ -0,0 +1,41 @@
+import { useEffect } from 'react';
+
+/**
+ * Prefetches the images that can be bottleneck for the pattern rendering.
+ */
+export const usePrefetchImage = () => {
+	const prefetchImage = ( url: string ) => {
+		const link = document.createElement( 'link' );
+		link.rel = 'prefetch';
+		link.as = 'image';
+		link.href = url;
+		document.head.appendChild( link );
+	};
+
+	const removeLinks = ( url: string ) => {
+		const link = document.querySelector( `link[href="${ url }"]` );
+		if ( link ) {
+			document.head.removeChild( link );
+		}
+	};
+
+	useEffect( () => {
+		const IMAGE_URLS = [
```

Comment:
[nit] We can define these images at the beginning of this file but it's fine to keep them here.

Reply:
+1 to this suggestion 🙂